src/prplmesh_add.lua

```diff
@@ -0,0 +1,124 @@
+# or
+#
+# (b) combination of their Contribution(s) with the work of authorship to which
+# such Contribution(s) was added by such copyright holder or contributor, if,
+# at the time the Contribution is added, such addition causes such combination
+# to be necessarily infringed. The patent license shall not apply to any other
+# combinations which include the Contribution.
+#
+# Except as expressly stated above, no rights or licenses from any copyright
+# holder or contributor is granted under this license, whether expressly, by
+# implication, estoppel or otherwise.
+#
+# DISCLAIMER
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+# USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+# NOTE
+#
+# This is part of a management middleware software package called MMX that was developed by Inango Systems Ltd.
+#
+# This version of MMX provides web and command-line management interfaces.
+#
+# Please contact us at Inango at support@inango-systems.com if you would like to hear more about
+# - other management packages, such as SNMP, TR-069 or Netconf
+# - how we can extend the data model to support all parts of your system
+# - professional sub-contract and customization services
+#
+################################################################################
+--]]
+
+require("prplmesh-be-utils")
+require("mmx/ing_utils")
+
+
+--[[
+    @brief  Function adds new instance to the Data Model represented by UBus.
+
+    @param  path String contains path to the object in Data Model.
+
+    @return MMX Style string for add object otherwise false.
+--]]
+local function add_object(path)
+
+    local result = call_ubus(path, "add", {})
+    if not result then
+        error("Failed to add: " .. tostring(path))
+        return false
+    end
+
+    local idx = 0
+    for k,v in pairs(result) do
+        if k == "index" then
+            idx = v
+            break
+        end
+    end
+
+    return tostring(ing.ResCode.SUCCESS) .. ";" .. tostring(ing.StatCode.OK) .. ";" .. tostring(idx) ..";"
+--add_object()
+end
+
+
+function main(args)
+
+    local ret = tostring(ing.ResCode.FAIL) .. ";" .. tostring(ing.StatCode.OK) .. ";"
+
+    if not args[1] then
+        error("Bad argument given: " .. tostring(args[1]))
+        return ret
+    end
+
+    local mmx = add_object(args[1])
```

Comment:
Add task to finish `add` script because this script must also accept and handle options line `-pname Parameter1Name -pvalue Parameter1Value -pname Parameter2Name -pvalue Parameter2Value ...`